src/main/java/co/rsk/federate/watcher/FederationWatcher.java

```diff
@@ -0,0 +1,113 @@
+    private Optional<Federation> retiringFederation = Optional.empty();
+
+    /**
+     * Constructs a new {@code FederationWatcher} with the specified RSK client.
+     *
+     * @param rsk the Ethereum client used to listen for new blocks on the RSK blockchain
+     */
+    public FederationWatcher(Ethereum rsk) {
+        this.rsk = rsk;
+    }
+
+    /**
+     * Starts the {@code FederationWatcher} by setting the {@code FederationProvider}
+     * and {@code FederationWatcherListener}, and begins listening for new blocks on
+     * the RSK blockchain.
+     *
+     * @param federationProvider        the provider used to obtain federation information (active and retiring federations)
+     * @param federationWatcherListener the listener that will be notified when the federations change
+     */
+    public void start(FederationProvider federationProvider, FederationWatcherListener federationWatcherListener) {
+        this.federationProvider = federationProvider;
+        this.federationWatcherListener = federationWatcherListener;
+     
+        rsk.addListener(new EthereumListenerAdapter() {
+            @Override
+            public void onBestBlock(org.ethereum.core.Block block, List<TransactionReceipt> receipts) {
+                // Updating state only when the best block changes still "works",
+                // since we're interested in finding out only when the active or retiring federation(s) changed.
+                // If there was a side chain in which any of these changed in, say, block 4500, but
+                // that side chain became main chain in block 7800, it would still be ok to
+                // start monitoring the new federation(s) on that block.
+                // The main reasons for this is that RSK nodes would have never reported the active or
+                // retiring federation(s) being different *before* the best chain change. Therefore,
+                // there should be no new Bitcoin transactions directed to these new addresses
+                // until this change effectively becomes a part of RSK's "reality".
+                // The case in which we go back and forth to and from a sidechain in which the
+                // federation effectively changed is still to be explored deeply, but the same reasoning
+                // should apply since going back and forth would trigger two federation changes.
+                // A client trying to send bitcoins to the new federation without waiting
+                // a good number of confirmations would be, essentially, "playing with fire".
+                logger.info("[updateState] New best block, updating state");
+                updateState();
+            }
+        });
+    }
+
+    /**
+     * Updates the current state of the federations by checking if the active or
+     * retiring federations have changed. If a federation change is detected, it notifies
+     * the {@code FederationWatcherListener}.
+     */
+    public void updateState() {
```

Comment:
That is a good question. It should not be with the changes now. [Before](https://github.com/rsksmart/powpeg-node/pull/327/files#diff-eee97682409d79c95cbd6586ebe72e6ea347c78fca3099803a616225986b5ce6L366) it made sense since it was calling it manually, but I think this is redundant and should already be doing it (at least btcreleaseclient doesn't do it). If that is okay, we can make this private. wdyt?